packages/golden_toolkit/lib/src/testing_tools.dart

```diff
@@ -159,12 +167,14 @@ Future<void> screenMatchesGolden(
   }
 
   final shouldSkipGoldenGeneration = skip ?? GoldenToolkit.configuration.skipGoldenAssertion();
+  final effectiveFileNameFactory = fileNameFactory ?? GoldenToolkit.configuration.fileNameFactory;
 
   final pumpAfterPrime = customPump ?? _onlyPumpAndSettle;
   /* if no finder is specified, use the first widget. Note, there is no guarantee this evaluates top-down, but in theory if all widgets are in the same 
   RepaintBoundary, it should not matter */
   final actualFinder = finder ?? find.byWidgetPredicate((w) => true).first;
-  final fileName = 'goldens/$goldenFileName.png';
+  final fileName = effectiveFileNameFactory(name);
+  await _primeImages(fileName, actualFinder);
```

Comment:
this line got duplicated (it's 3 lines down) from the merge conflict. Can be nuked